BibBuddy/app/src/main/java/de/bibbuddy/BookAddManuallyFragment.java

```diff
@@ -0,0 +1,279 @@
+package de.bibbuddy;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The BookAddManuallyFragment is responsible for adding a book manually to a shelf.
+ *
+ * @author Claudia Schönherr
+ */
+public class BookAddManuallyFragment extends Fragment {
+  private final AddBookManuallyListener listener;
+  List<Author> authorList;
+  private boolean validInput;
+  private Long shelfId;
+  private String shelfName;
+  private int redColor;
+  private int greenColor;
+  private Book book;
+
+  public BookAddManuallyFragment(AddBookManuallyListener listener) {
+    this.listener = listener;
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                           @Nullable Bundle savedInstanceState) {
+    // Called to have the fragment instantiate its user interface view.
+    View view = inflater.inflate(R.layout.fragment_book_add_manually, container, false);
+
+    setupButtons(view);
+    setupInput(view);
+
+    Bundle bundle = this.getArguments();
+    shelfName = bundle.getString(LibraryKeys.SHELF_NAME);
+    shelfId = bundle.getLong(LibraryKeys.SHELF_ID);
+
+    redColor = getResources().getColor(R.color.alert_red);
+    greenColor = getResources().getColor(R.color.green);
+
+    ((MainActivity) getActivity()).updateHeaderFragment(getString(R.string.add_book));
+
+    return view;
+  }
+
+  private void setupInput(View view) {
+    view.findViewById(R.id.book_form_isbn_input).requestFocus();
+    InputMethodManager imm =
```

Comment:
```suggestion
    InputMethodManager inputManager =
```

Reply:
imm is not a good variable name, because it is not comprehensible. Better write inputManager or something like that.